src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -570,7 +573,7 @@ internal virtual void AddHeaders(HttpHeaders sourceHeaders)
                     else
                     {
                         Debug.Assert(sourceValue is string);
-                        _headerStore.Add(header.Key, _forceHeaderStoreItems ? new HeaderStoreItemInfo { RawValue = sourceValue } : sourceValue);
+                        _headerStore.Add(header.Key, _forceHeaderStoreItems ? new HeaderStoreItemInfo { RawValue = sourceValue, OriginalRawValue = sourceValue } : sourceValue);
```

Comment:
I don't understand why the existing code is not forcing parsing in this case. We force parsing above, so why not here? Seems inconsistent.

Reply:
Sorry, I don't know right now. It would take more time to investigate. Anyways, this PR doesn't change this logic.